src/lib/statistics/column_statistics.cpp

```diff
@@ -0,0 +1,444 @@
+#include "column_statistics.hpp"
+
+#include <sstream>
+
+#include "resolve_type.hpp"
+#include "table_statistics.hpp"
+#include "type_cast.hpp"
+
+namespace opossum {
+
+template <typename ColumnDataType>
+ColumnStatistics<ColumnDataType>::ColumnStatistics(const float null_value_ratio, const float distinct_count,
+                                                   const ColumnDataType min, const ColumnDataType max)
+    : AbstractColumnStatistics(data_type_from_type<ColumnDataType>(), null_value_ratio, distinct_count),
+      _min(min),
+      _max(max) {
+  Assert(null_value_ratio >= 0.0f && null_value_ratio <= 1.0f, "NullValueRatio out of range");
```

Comment:
`DebugAssert`?

Reply:
when I started running release builds I started valuing Assert over the segfault/whatnot I'd otherwise get. So I started using Assert if a path is not performance critical.

Reply:
And this one certainly isn't